Packages/com.unity.inputsystem/InputSystem/Plugins/InputForUI/InputSystemProvider.cs

```diff
@@ -640,9 +640,7 @@ void UnregisterActions(Configuration cfg)
             m_InputActionAsset.Disable();
 
             // The Next/Previous action is not part of the input actions asset
-            UnregisterNextPreviousAction();
-
-            UnityEngine.Object.Destroy(m_InputActionAsset); // TODO check if this is ok
+            UnregisterNextPreviousAction();            UnityEngine.Object.DestroyImmediate(m_InputActionAsset); // TODO check if this is ok
```

Comment:
Seems like formatting in this file is off, did you run formatter, /format.ps1 or /format.sh?

I noticed there is also another DestroyImmediate in this file. Why do you need DestroyImmediate here? Notice that its generally recommended against (See API docs) and should only be used in edit mode, so if we need it use

#if UNITY_EDITOR
... destroy immediate ....
#else
... destroy
#endif

However, I wonder why we can't use Destroy in both places in this file?

Reply:
Huh that's weird, It didn't look like that in Visual Studio to me when I was editing it 😅 
>I noticed there is also another DestroyImmediate in this file. Why do you need DestroyImmediate here? Notice that its generally recommended against (See API docs) and should only be used in edit mode, so if we need it use

I'm following instructions from Adrien for this bug: https://jira.unity3d.com/browse/ISXB-677 Sorry, I forgot to add the work in progress tag. I'm currently setting up the mentioned SRP tests to see if my changes actually fix the error in question

Reply:
Would you be able to consult Adrien regarding if #ifdef is needed or not? Basically based on remarks here: https://docs.unity3d.com/ScriptReference/Object.DestroyImmediate.html